micromouse/RobotIO.cpp

```diff
@@ -184,21 +221,74 @@ namespace Micromouse
 		leftMotor.brake();
 		rightMotor.brake();
 #endif
+*/
+
 	}
 
-	// ## NOT YET IMPLEMENTED ##
-	void RobotIO::moveForward()
-	{
 
 
+	void RobotIO::moveForward(float millimeters)
+	{		//centimeters represents how much farther the bot needs to travel.
+			//The function will loop until centimeters is within DISTANCE_TOLERANCE
+
+		leftMotor.setMaxSpeed(0.25f);
+		rightMotor.setMaxSpeed(0.25f);
+		leftMotor.resetCounts();
+		rightMotor.resetCounts();
+
+		distPID.start(millimeters);
+		headingPID.start(estimateHeadingError());
+
+		while (millimeters > DISTANCE_TOLERANCE || millimeters < -DISTANCE_TOLERANCE)
+		{
+
+			//Get distance traveled in cm since last cycle (average of two encoders)
+			float traveled = leftMotor.resetCounts() + rightMotor.resetCounts(); //resetCounts() also returns counts
+			traveled /= 2;
+			traveled /= COUNTS_PER_MM;
+
+			//Decrease distance to go by the estimated amount traveled
+			millimeters -= traveled;
+
+			//Get correction speed
+			float speed = distPID.getCorrection(millimeters);
+
+			//Get rotational correction speed
+			float rotSpeed = headingPID.getCorrection(estimateHeadingError());
```

Comment:
ohhh i finally get what the function does... I think it could have a better name.. but i cant think of any.